### ISSUE TITLE
Optimize YUV unlit shader

### ISSUE THREAD

**Issue by eirikhollis** [microsoft/MixedReality-WebRTC#110]
The precision in the YUV shader can be argued to be unnecessarily high, and could benefit with lower precision without losing quality (at least not as far as I can see). 
Added code for the YUVFeedShaderUnlit shader to work with "Single Pass Instanced" stereo rendering mode and changed float4 variables to fixed3 for performance improvements.
The other shaders remain unchanged for now.

Fixed syntax, as Unity describes it:
`Lowest precision fixed point value. Generally 11 bits, with a range of –2.0 to +2.0 and 1/256th precision.`
Changed therefore the amount of decimals in the YUV to RGB conversion to 3.

```
Shader "Video/YUVFeedShader (unlit)"
{
    Properties
    {
        [Toggle(MIRROR)] _Mirror("Horizontal Mirror", Float) = 0
        [HideInEditor][NoScaleOffset] _YPlane("Y plane", 2D) = "white" {}
        [HideInEditor][NoScaleOffset] _UPlane("U plane", 2D) = "white" {}
        [HideInEditor][NoScaleOffset] _VPlane("V plane", 2D) = "white" {}
    }
    SubShader
    {
        Pass
        {
            CGPROGRAM
            #pragma vertex vert
            #pragma fragment frag
            #pragma multi_compile __ MIRROR

            #include "UnityCG.cginc"

            struct appdata
            {
                float4 vertex : POSITION;
                float2 uv : TEXCOORD0;

                UNITY_VERTEX_INPUT_INSTANCE_ID
            };

            struct v2f
            {
                float2 uv : TEXCOORD0;
                float4 vertex : SV_POSITION;

                UNITY_VERTEX_OUTPUT_STEREO
            };

            v2f vert (appdata v)
            {
                v2f o;

                UNITY_SETUP_INSTANCE_ID(v);
                UNITY_INITIALIZE_OUTPUT(v2f, o);
                UNITY_INITIALIZE_VERTEX_OUTPUT_STEREO(o);

                o.vertex = UnityObjectToClipPos(v.vertex);
                o.uv = v.uv;
#if !UNITY_UV_STARTS_AT_TOP
                o.uv.y = 1 - v.uv.y;
#endif
#ifdef MIRROR
                o.uv.x = 1 - v.uv.x;
#endif
                return o;
            }

            sampler2D _YPlane;
            sampler2D _UPlane;
            sampler2D _VPlane;

            fixed3 yuv2rgb(fixed3 yuv)
            {
                // The YUV to RGB conversion, please refer to: http://en.wikipedia.org/wiki/YUV
                // Y'UV420p (I420) to RGB888 conversion section.
                fixed y_value = yuv[0];
                fixed u_value = yuv[1];
                fixed v_value = yuv[2];
                fixed r = y_value + 1.371 * (v_value - 0.5);
                fixed g = y_value - 0.698 * (v_value - 0.5) - (0.338 * (u_value - 0.5));
                fixed b = y_value + 1.732 * (u_value - 0.5);
                return fixed3(r, g, b);
            }

            fixed3 frag (v2f i) : SV_Target
            {
                fixed3 yuv;
                yuv.x = tex2D(_YPlane, i.uv).r;
                yuv.y = tex2D(_UPlane, i.uv).r;
                yuv.z = tex2D(_VPlane, i.uv).r;
                return yuv2rgb(yuv);
            }
            ENDCG
        }
    }
}
```

**Comment by djee-ms** [comment]
Completely agreed on stereo rendering, but a bit skeptical on the use of `fixed` for color conversion. I am not an expert on those things by all means, but I thought `half` and `float` were the only 2 hardware types for GPU. It's also not clear from the docs if this `fixed` type is only for texture storage or also for arithmetic operations. So it feels like using `half` might get us some performance improvement because it's backed by shorter hardware registers on the GPU, but using `fixed` and reducing the precision of the conversion constants might just hurt precision compared to `half` without changing anything to performance since the hardware registers used for the arithmetic operations are still `half` (16 bits). Thoughts?

**Comment by eirikhollis** [comment]
Found this on Unity Docs
`Fixed precision is generally only useful for older mobile GPUs. Most modern GPUs (the ones that can run OpenGL ES 3 or Metal) internally treat fixed and half precision exactly the same.`
So you might be correct, there might not be a difference between half and fixed.

Did some color visual tests, as well as brightness/darkness between the fixed and float, with and without all decimals on constants, and in my eyes it look completely the same (even though it might not be on a microscopic level), so I don't think there is any need for more precision than 3 decimals.

**Comment by djee-ms** [comment]
Fixed with 2f93b05df3bd9a2f8e4cc7d7d4e74641d59450f9 and 3036c8857a43dcd1ba7a250a8cbffabdeacb0bdf. I talked with rendering engineers who confirmed that `half` is good enough because:
- Most GPUs found in desktop machines don't support low-precision, only medium and high.
- Only some mobile devices support low-precision float in hardware, but the shader is bandwidth-limited anyway so this would have no effect.
